spec/stripe_tester_spec.rb

```diff
@@ -15,27 +15,44 @@
       StripeTester.verify_ssl = nil
     end
 
-    it "#load_template should return hash" do
-      result = StripeTester.load_template(:invoice_created)
-
-      expect(result).to be_a_kind_of(Hash)
-    end
-
-    it "#load_template should return correct callback type" do
-      type = "invoice_created"
-
-      returned_hash = StripeTester.load_template(type)
-      returned_type = returned_hash["type"]
-
-      returned_type.sub!('.', '_')
-
-      expect(returned_type).to eq(type)
-    end
-
-    it "#load_template should raise an exception when invalid event is given" do
-      type = "incorrect_type"
-
-      expect { StripeTester.load_template(type) }.to raise_error
+    describe "#load_template" do
+      it "should return hash" do
+        result = StripeTester.load_template(:invoice_created)
+
+        expect(result).to be_a_kind_of(Hash)
+      end
+    
+      context "Hash with_indifferent_access supported" do
+        it "should call with_indifferent_access on Hash" do
+          allow_any_instance_of(Hash).to receive(:with_indifferent_access)
+          expect_any_instance_of(Hash).to receive(:with_indifferent_access)
+          StripeTester.load_template(:invoice_created)
+        end
+      end
+    
+      context "Hash with_indifferent_access not supported" do
+        it "should not call with_indifferent_access" do
+          allow_any_instance_of(Hash).to receive(:respond_to?).and_return(false)
```

Comment:
Suggested refactor:

```ruby
allow_any_instance_of(Hash).to receive(:respond_to?).with(:with_indifferent_access).and_return(false)
```

I think stubbing `with_indifferent_access` specifically is much better, since stubbing `respond_to?` with any argument would cause unexpected bugs, e.g. something else calls `respond_to?` inside `StripeTester` and expects back `true`.